### PR TITLE
feat: implemented Login form

### PR DESCRIPTION
**Pull request by prabhat-wednesday**
### Ticket Link

---

### Related Links

---

### Description
- Add Login Form Container 
---

### Steps to Reproduce / Test

---

---

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---

### CODE REVIEW

File: app/containers/LoginContainer/tests/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { renderProvider } from '@app/utils/testUtils';
+import LoginContainer from '..';
+
+describe('<LoginContainer tests', () => {
```
Comment (christinittyjohn) [inline app/containers/LoginContainer/tests/index.test.tsx:8]: Please add tests for the functionality.

**Review by christinittyjohn — CHANGES_REQUESTED** [review]

File: app/containers/LoginContainer/index.tsx [hunk 2]
```diff
@@ -13,14 +15,16 @@ const FormContainer = styled.div`
...
-  width: 100%;
-`;
-
-interface LoginContainerProps {
+export interface LoginContainerProps {
   dispatchUserData: (payload: any) => AnyAction;
 }
 
+const FormInputArray = [
+  { label: 'Username', name: 'username', message: 'Please input your username!' },
```
Comment (christinittyjohn) [inline app/containers/LoginContainer/index.tsx]: Please add support for translations

File: app/components/FormInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+`;
+
+interface FormInputProps {
+  label: string;
+  name: string;
+  message: string;
+}
+const FormInput = ({ label, name, message }: FormInputProps) => {
+  return (
+    <Form.Item label={label} name={name} rules={[{ required: true, message }]}>
```
Comment (christinittyjohn) [inline app/components/FormInput/index.tsx]: This sets every field as required.

**Review by christinittyjohn — CHANGES_REQUESTED** [review]

File: app/components/FormInput/index.tsx (hunk 3 above)
Comment (prabhat-wednesday) [reply app/components/FormInput/index.tsx]: shouldn't  every fields to be required

File: app/containers/App/selector.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { createSelector } from 'reselect';
+import get from 'lodash-es/get';
+import { initialState } from '../LoginContainer/reducer';
+
+export const selectLoginState = (state: any) => state.loginReducer || initialState;
```
Comment (christin-is-buildding) [inline app/containers/App/selector.ts]: please move this to the login container

**Review by christin-is-buildding — CHANGES_REQUESTED** [review]
please improve the tests in FormInput

File: app/translations/en.json [hunk 5]
```diff
@@ -12,5 +12,14 @@
   "type_label": "<b>Type</b> {type}",
   "ships": "Ships ",
   "rocket": "Rocket ",
-  "song_search_default": "Search your favourite singer"
+  "song_search_default": "Search your favourite singer",
+  "label_1": "Username",
```
Comment (christinittyjohn) [inline app/translations/en.json]: please use descriptive keys

**Review by christinittyjohn — CHANGES_REQUESTED** [review]

**Review by christinittyjohn — CHANGES_REQUESTED** [review]
please change translation keys for form labels.

**Review by christinittyjohn — APPROVED** [review]

**Pull request merged**